src/mutation_data.cpp

```diff
@@ -567,13 +567,51 @@ std::string mutation_branch::desc() const
     return raw_desc.translated();
 }
 
+static bool has_cyclic_dependency( const trait_id &mid, std::vector<trait_id> already_visited )
+{
+    if( contains_trait( already_visited, mid ) ) {
+        return true; // Circular dependency was found.
+    }
+
+    already_visited.push_back( mid );
+
+    // Perform Depth-first search
+    for( const trait_id &current_mutation : mid->prereqs ) {
+        if( has_cyclic_dependency( current_mutation->id, already_visited ) ) {
+            return true;
+        }
+    }
+
+    for( const trait_id &current_mutation : mid->prereqs2 ) {
+        if( has_cyclic_dependency( current_mutation->id, already_visited ) ) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static void check_has_cyclic_dependency( const trait_id &mid )
+{
+    std::vector<trait_id> already_visited;
+    if( ::has_cyclic_dependency( mid, already_visited ) ) {
```

Comment:
The double colon is used to refer to global variables. So it is necessary for `::check_consistency` as there are two, `mutation_branch::check_consistency` and `static void check_consistency` and we want the latter here (the global one). It's not necessary for `check_has_cyclic dependency`.